atomic_reactor/plugin.py

```diff
@@ -258,10 +274,21 @@ def run(self, keep_going=False):
                 logger.exception("failed to save plugin duration")
 
             self.plugins_results[plugin_class.key] = plugin_response
+
+            if plugin_successful and buildstep_phase:
+                logger.debug('stopping further execution of plugins '
+                             'after first successful plugin')
+                break
+
         if len(failed_msgs) == 1:
             raise PluginFailedException(failed_msgs[0])
         elif len(failed_msgs) > 1:
             raise PluginFailedException("Multiple plugins raised an exception: " + str(failed_msgs))
+
+        if (plugins_skipped == len(self.plugins_conf)) and buildstep_phase:
```

Comment:
that would work, except for when you would also set is_allowed_to_fail on plugin before last failed,

this way it really counts only when it raises InappropriateBuildStepError,
or when plugin doesn't exist